plugins/utils/notify.nix

```diff
@@ -7,67 +7,141 @@
 with lib; let
   cfg = config.plugins.notify;
   helpers = import ../helpers.nix {inherit lib;};
-  icon = mkOption {
-    type = types.nullOr types.str;
-    default = null;
-  };
 in {
-  options.plugins.notify = {
-    enable = mkEnableOption "notify";
+  options.plugins.notify =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "nvim-notify";
 
-    package = helpers.mkPackageOption "notify" pkgs.vimPlugins.nvim-notify;
+      package = helpers.mkPackageOption "nvim-notify" pkgs.vimPlugins.nvim-notify;
 
-    stages = mkOption {
-      type = types.nullOr (types.enum ["fade_in_slide_out" "fade" "slide" "static"]);
-      description = "Animation style";
-      default = null;
-    };
-    timeout = mkOption {
-      type = types.nullOr types.int;
-      description = "Default timeout for notifications";
-      default = null;
-    };
-    backgroundColour = mkOption {
-      type = types.nullOr types.str;
-      description = "For stages that change opacity this is treated as the highlight between the window";
-      default = null;
-    };
-    minimumWidth = mkOption {
-      type = types.nullOr types.int;
-      description = "Minimum width for notification windows";
-      default = null;
-    };
-    icons = mkOption {
-      type = types.nullOr (types.submodule {
-        options = {
-          error = icon;
-          warn = icon;
-          info = icon;
-          debug = icon;
-          trace = icon;
+      level =
+        helpers.defaultNullOpts.mkNullable
+        (
+          with types;
+            oneOf [
+              ints.unsigned
+              helpers.rawType
+              str
+            ]
+        )
+        ''{__raw = "vim.log.levels.INFO";}''
+        "Minimum log level to display. See `vim.log.levels`.";
+
+      timeout = helpers.defaultNullOpts.mkUnsignedInt 5000 "Default timeout for notification.";
+
+      maxWidth = helpers.mkNullOrOption (with types; either ints.unsigned helpers.rawType) ''
+        Max number of columns for messages.
+      '';
+
+      maxHeight = helpers.mkNullOrOption (with types; either ints.unsigned helpers.rawType) ''
+        Max number of lines for a message.
+      '';
+
+      stages =
```

Comment:
It seems that the stages should either be a string or an array of functions, do we leave the array of functions to do in the raw type or do we use listOf rawType ?

Reply:
Good point, I chose a `listOf str` that are `mkRaw`ed afterwards.